### DISCUSSION TITLE
ExtrapolationLineSeries Documentation

### DISCUSSION THREAD

**Discussion by dknippel** [oxyplot/oxyplot#1907] [Q&A] [answered]
Hello,

I am developing a WPF application that uses the OxyPlot control.  I have a specific need to use the newly released **_ExtrapolationLineSeries_** feature.  However, I cannot find any documentation on it other than the original issue #1740 [Line Series with style for extrapolated/intropolated regions.](https://github.com/oxyplot/oxyplot/issues/1740)

Is there any documentation/examples for said new feature?

Thank you,
Dennis

**Comment by VisualMelon** [comment, marked as answer]
No, I'm afraid there is no documentation for it. There are examples in the Example Library: https://github.com/oxyplot/oxyplot/blob/develop/Source/Examples/ExampleLibrary/Series/ExtrapolationLineSeriesExamples.cs

**Reply by dknippel** [reply]
Thank you for your reply.  These examples are helpful.  However, I was wondering what the intent/use of the `BrokenLineStyle` is.  Could you provide some insight?

Also, is it safe to assume that `Intervals` can now overlap and do not need to be in ascending order?

**Reply by VisualMelon** [reply]
Looks like `BrokenLineStyle` isn't used; it's just inherited from `LineSeries`.

Yes, it will process the intervals in any order/overlapping arrangement.

**Reply by dknippel** [reply]
Again, thank you for your information.

So, are the `Intervals` intended to be ranges that are rendered using the

**_Extrapolation_** members (ExtrapolationColor, ExtrapolationLineStyle, etc.)

OR

the default members (Color, LineStyle, etc.)

**Reply by VisualMelon** [reply]
Intervals are the extrapolated regions.

**Reply by dknippel** [reply]
Ah, I see.  I was using the intervals for the default members.  Will switch them and see if that results in being fully joyed...

**Reply by dknippel** [reply]
I'm still a bit confused about the `ExtrapolationLineSeries` (or perhaps the `LineSeries` in general).  I have assumed that the order in which I add `DataPoint` 's to a `Points` collection does not matter.  IOW, the following scenario should render the same plot:
```
PlotModel PM = new PlotModel;

Series seriesA = new ExtrapolationLineSeries();
PM.Series.Add(seriesA);
DataPoint dpA1 = new DataPoint(1, 1); seriesA.Points.Add(dpA1);
DataPoint dpA2 = new DataPoint(2, 4);seriesA.Points.Add(dpA2);
DataPoint dpA3 = new DataPoint(3, 3);seriesA.Points.Add(dpA3);

Series seriesB = new ExtrapolationLineSeries();
PM.Series.Add(seriesB);
DataPoint dpB2 = new DataPoint(2, 4);seriesB.Points.Add(dpB2);
DataPoint dpB1 = new DataPoint(1, 1);seriesB.Points.Add(dpB1);
DataPoint dpB3 = new DataPoint(3, 3);seriesB.Points.Add(dpB3);

PM.InvalidatePlot(true);

```

However, I've found that the first series renders a contiguous line that rises from left to right  whereas the second series renders zig-zag that falls from right to left, then left to right.  Is this order dependency something unique to `ExtrapolationLineSeries` ?  OR... is it perhaps the manner in which I invoke `InvalidatePlot()` ?

UPDATE:  I have confirmed that there does indeed seem to be a dependency on the sequential order in which `DataPoint` 's are added to the `Points` collection, regardless of when `InvalidatePlot()` is called.  Is this by design?  If so, I will need to find another solution seeing how my data points are generated in a non-deterministic/non-sequential manner.

**Reply by VisualMelon** [reply]
Yes, the points are rendered in the order they appear in the `Points` collection: this is by design. `InvalidatePlot` won't change anything about how it is rendered, only when. If you need them sorted, consider an in-place sort when invalidating. It's not ideal, but I'd be surprised if it was slow compared to rendering for any volume of points that the library can actually handle.
